### PR TITLE
Client IITO Implementation 

### PR DESCRIPTION
**Pull request by agrimmanchanda**
# Summary

Includes all tests and agent strategy implementation of IITO:

GetGiftRequests()
GetGiftOffers()
GetGiftResponses()
UpdateGiftInfo()
SentGift()
ReceivedGift()
DecideGiftAmount()

## Additional Information

This PR is in addition to the previous (accidental) merged PR #149 into agent-strategy from agent-strategy-roberts-IITO-agrim. 

Please review entire **clientiitoiifo.go** (if possible). 

Also, I have also added updatetrustScore() and updateTheirTrustScore() function in the **StartOfTurn()** function so it will update trust scores at the **beginning** of each turn with changes stored in trustMapAgg and theirTrustMapAgg. 

## Test Plan

Tests included in **clientiitoiifo_test.go**

### CODE REVIEW

File: internal/clients/team3/clientgeneral.go [hunk 1]
```diff
@@ -50,8 +54,8 @@ func (c *client) Initialise(serverReadHandle baseclient.ServerReadHandle) {
 	c.ourSpeaker = speaker{c: c}
 	c.ourJudge = judge{c: c}
 	c.ourPresident = president{c: c}
-  
-  c.initgiftOpinions()
+
+	c.initgiftOpinions()
```
Comment (Neelesh99) [inline internal/clients/team3/clientgeneral.go:58]: Padding your LOC I see

File: internal/clients/team3/clientiitoiifo.go [hunk 2]
```diff
@@ -72,78 +72,78 @@ func findAvgExclMinMax(Requests shared.GiftRequestDict) shared.GiftRequest {
 	return shared.GiftRequest(float64(sum) / float64(len(shared.TeamIDs)))
 }
 
+// sigmoidAndNormalise returns the normalised number between 0 - 1 based on the
+// trust score between 0 - 100.
+func (c *client) sigmoidAndNormalise(island shared.ClientID) shared.GiftOffer {
```
Comment (Neelesh99) [inline internal/clients/team3/clientiitoiifo.go:218]: Damn boi, this is sick

**Review by Neelesh99 — APPROVED** [review]
Looks good to me

File: internal/clients/team3/clientiitoiifo.go (hunk 2 above)
Comment (tharusha-a) [reply internal/clients/team3/clientiitoiifo.go:218]: WOAHHHHH

**Review by tharusha-a — APPROVED** [review]

**Pull request merged**